src/bdecode.cpp

```diff
@@ -102,13 +102,25 @@ namespace {
 
 	struct stack_frame
 	{
-		stack_frame() : token(0), state(0) {}
-		explicit stack_frame(int const t): token(std::uint32_t(t)), state(0) {}
+		stack_frame() : m_token_state(0) {}
```

Comment:
what's the motivation for this change? it seems to increase complexity, and I don't know what it buys